vault/resource_database_secret_backend_connection.go

```diff
@@ -763,6 +767,17 @@ func connectionStringResource(config *connectionStringConfig) *schema.Resource {
 			Description: "The root credential password used in the connection URL",
 			Sensitive:   true,
 		}
+		res.Schema[consts.FieldPasswordWO] = &schema.Schema{
```

Comment:
Do we need a `ConflictsWith` on these?

Reply:
It doesn't look like we  can enforce `ConflictsWith` and `RequiredWith` on these fields since they are part of a nested `schema.Resource` block 😞  I instead added explicit conditionals to ensure that `password_wo` will only be read if `password_wo_version` is provided, and if the conflicting legacy field `password` is provided, the write-only attributes are ignored. A log message is also dispatched to the user, instructing them to move to using the new write-only attributes in that case